### PR TITLE
Freddy/module ownership transfer

### PR DESCRIPTION
**Pull request by freddyli7**
## Description

- Add module ownership transfer tx and updated anteHandler

### CODE REVIEW

File: scripts/cmd.sh [hunk 1]
```diff
@@ -16,12 +16,15 @@ chainlinkd query chainlink getLatestFeedData "testfeedid1" --chain-id testchain
...
 chainlinkd keys list --keyring-backend test
 
+# List all module owner
+chainlinkd query chainlink getModuleOwnerList --chain-id testchain -o json
+
 # Add new module owner
-chainlinkd tx chainlink addModuleOwner "cosmos12uxqzq7aae5ew2rksy0cl0cua7e75cu5t3rjxf" "cosmospub1addwnpepqw2xec9wutvxdhvgke029v3hx97jdpzqpr6d33jx62drd92qjxxrsm9t7q5" --from cosmos1mxea03k4y8n7c4fyeseu3c889ejw7kncwhp0uv --keyring-backend test --chain-id testchain
+chainlinkd tx chainlink addModuleOwner "cosmos1wjthz4kmkcusava94f55pg06cqrlxm889udgjn" "cosmospub1addwnpepq00vr93dx4k88rpfupm5wxv50nmq69chxlfal279cexxjy8yl29dc2kqqqn" --from alice --keyring-backend test --chain-id testchain
+
+# module ownership transfer
```
Comment (araskachoi) [inline scripts/cmd.sh:28]: it is a bit unclear what this test is for. can you add as a comment the transfer that should occur?
ex: `# module ownership transfer from Alice to Bob`

Comment (araskachoi) [reply scripts/cmd.sh:28]: the cosmos addr and pubkey are too arbitrary to understand here

File: x/chainlink/keeper/keeper.go [hunk 2]
```diff
@@ -75,11 +75,11 @@ func (k Keeper) SetFeedData(ctx sdk.Context, feedData *types.MsgFeedData) (int64
 		RoundId:               roundId,
 	}
 
-	feedStore := ctx.KVStore(k.feedStoreKey)
+	feedDateStore := ctx.KVStore(k.feedDataStoreKey)
```
Comment (araskachoi) [inline x/chainlink/keeper/keeper.go:78]:
```suggestion
	feedDataStore := ctx.KVStore(k.feedDataStoreKey)
```

File: x/chainlink/keeper/keeper.go [hunk 3]
```diff
@@ -75,11 +75,11 @@ func (k Keeper) SetFeedData(ctx sdk.Context, feedData *types.MsgFeedData) (int64
...
 
 	f := k.cdc.MustMarshalBinaryBare(&finalFeedDataInStore)
 
-	feedStore.Set(types.KeyPrefix(types.FeedDataKey+feedData.FeedId+fmt.Sprintf("%d", roundId)), f)
+	feedDateStore.Set(types.KeyPrefix(types.FeedDataKey+feedData.FeedId+fmt.Sprintf("%d", roundId)), f)
```
Comment (araskachoi) [inline x/chainlink/keeper/keeper.go:82]:
```suggestion
	feedDataStore.Set(types.KeyPrefix(types.FeedDataKey+feedData.FeedId+fmt.Sprintf("%d", roundId)), f)
```

File: x/chainlink/types/msg.go [hunk 4]
```diff
@@ -74,32 +73,36 @@ func NewModuleOwner(assigner, address sdk.Address, pubKey []byte) *ModuleOwner {
...
 	return sdk.MustSortJSON(bz)
 }
 
-func (m *ModuleOwner) GetSigners() []sdk.AccAddress {
+func (m *MsgModuleOwner) GetSigners() []sdk.AccAddress {
 	return []sdk.AccAddress{sdk.AccAddress(m.AssignerAddress)}
 }
 
-type ModuleOwners []*ModuleOwner
+func (m *MsgModuleOwner) GetMsgs() []githubcosmossdktypes.Msg {
```
Comment (araskachoi) [inline x/chainlink/types/msg.go:98]: not quite sure what this is used for,,

**Review by araskachoi — APPROVED** [review]

File: scripts/cmd.sh (hunk 1 above)
Comment (freddyli7) [reply scripts/cmd.sh:28]: ok

File: x/chainlink/types/msg.go (hunk 4 above)
Comment (freddyli7) [reply x/chainlink/types/msg.go:98]: this is used in anteHandler to parse txs from Msg and do type casting

File: scripts/cmd.sh (hunk 1 above)
Comment (freddyli7) [reply scripts/cmd.sh:28]: the address and key are always different when adding a new key

**Pull request merged**